### ISSUE TITLE
BCELoss for binary classification model

### ISSUE THREAD

**Issue by L-M-Sherlock** [tracel-ai/burn#672]
### Feature description

BCELoss: https://pytorch.org/docs/stable/generated/torch.nn.BCELoss.html

### Feature motivation

The current loss function is CrossEntropyLoss, which has already included a softmax layer inside. Its input is expected to contain the unnormalized logits for each class.

However, my model's output is probability. The softmax layer inside the CrossEntropyLoss will process the probability in an unexpected way. So the weights couldn't be optimized as expected.

- https://github.com/open-spaced-repetition/fsrs-optimizer-burn/issues/17

### Suggest a Solution

My implementation:

```rust
    fn bceloss(&self, probabilities: Tensor<B, 1>, targets: Tensor<B, 1>) -> Tensor<B, 1> {
        let loss: Tensor<B, 1> =
            targets.clone() * probabilities.clone().log() + (-targets + 1) * (-probabilities + 1).log();
        loss.mean().neg()
    }
```

This function works for my model. I hope it would be helpful.

**Comment by laggui** [comment]
We have `BinaryCrossEntropyLoss` now with #755 #1571
